### ISSUE TITLE
net-wireless/srsran-9999

### ISSUE THREAD

**Issue by richardredditch** [pentoo/pentoo-overlay#1491]
I have a problem here to complete compilation. To do so I have to mask parts of the ebuild like this:

        # "fix" "auto-detection" from use flags, this is probably horrible
        if use cpu_flags_x86_sse; then
                append-cflags "-DLV_HAVE_SSE -mfpmath=sse -msse4.1"
                append-cxxflags "-DLV_HAVE_SSE -mfpmath=sse -msse4.1"
        fi
        #if use cpu_flags_x86_avx; then
        #       append-cflags "-DLV_HAVE_AVX -mavx"
        #       append-cxxflags "-DLV_HAVE_AVX -mavx"
        #fi
        #if use cpu_flags_x86_avx2; then
        #       append-cflags "-DLV_HAVE_AVX2 -mavx2"
        #       append-cxxflags "-DLV_HAVE_AVX2 -mavx2"
        #fi
        #if use cpu_flags_x86_avx512f; then
        #       append-cflags "-DLV_HAVE_AVX512 -mavx512f"
        #       append-cxxflags "-DLV_HAVE_AVX512 -mavx512f"
        #fi
        if use cpu_flags_x86_fma3; then
                append-cflags "-DLV_HAVE_FMA -mfma"
                append-cxxflags "-DLV_HAVE_FMA -mfma"
        fi
        cmake_src_configure
}

**Comment by ZeroChaos-** [comment]
No build log means no build failure

**Comment by richardredditch** [comment]
missing build log: http://0x0.st/oCjh.txt

**Comment by ZeroChaos-** [comment]
That I can work with.  Can you share /proc/cpuinfo please?  I just need one cpu worth

**Comment by richardredditch** [comment]
after:

        #       append-cflags "-DLV_HAVE_AVX -mavx"
        #       append-cxxflags "-DLV_HAVE_AVX -mavx"
        #fi

build log:  http://0x0.st/oCjF.txt

**Comment by ZeroChaos-** [comment]
It's avx512f causing the failure. The other removals appear unnecessary

**Comment by richardredditch** [comment]
true, with:

        # "fix" "auto-detection" from use flags, this is probably horrible
        if use cpu_flags_x86_sse; then
                append-cflags "-DLV_HAVE_SSE -mfpmath=sse -msse4.1"
                append-cxxflags "-DLV_HAVE_SSE -mfpmath=sse -msse4.1"
        fi
        if use cpu_flags_x86_avx; then
                append-cflags "-DLV_HAVE_AVX -mavx"
                append-cxxflags "-DLV_HAVE_AVX -mavx"
        fi
        #if use cpu_flags_x86_avx2; then
        #       append-cflags "-DLV_HAVE_AVX2 -mavx2"
        #       append-cxxflags "-DLV_HAVE_AVX2 -mavx2"
        #fi
        #if use cpu_flags_x86_avx512f; then
        #       append-cflags "-DLV_HAVE_AVX512 -mavx512f"
        #       append-cxxflags "-DLV_HAVE_AVX512 -mavx512f"
        #fi
        if use cpu_flags_x86_fma3; then
                append-cflags "-DLV_HAVE_FMA -mfma"
                append-cxxflags "-DLV_HAVE_FMA -mfma"
        fi
        cmake_src_configure
}

build log:  http://0x0.st/oCe-.txt

**Comment by ZeroChaos-** [comment]
Thanks.  I had almost the same non-descript failure with avx2, which is how it got commented out.  I will look into this further, thanks for the report.

**Comment by edwardgalligan** [comment]
Looks like the change made to the srsran-22.10 ebuild has broken the checksum check when downloading the tarball at https://codeload.github.com/srsran/srsRAN_4G/tar.gz/refs/tags/release_22_10

Change: https://github.com/pentoo/pentoo-overlay/commit/b59b67bbcd8a7bc9695904390348a106461bfa59

Fwiw I haven't (yet) successfully built srsran-22.10 & there are other issues reported with this version @ https://github.com/srsran/srsRAN_4G/issues/1111 & @ https://github.com/srsran/srsRAN_4G/issues/1116 so very possible the build fill still fail after passing the tarball checksum check.

**Comment by ZeroChaos-** [comment]
> Looks like the change made to the srsran-22.10 ebuild has broken the checksum check when downloading the tarball at https://codeload.github.com/srsran/srsRAN_4G/tar.gz/refs/tags/release_22_10
> 
> Change: https://github.com/pentoo/pentoo-overlay/commit/b59b67bbcd8a7bc9695904390348a106461bfa59
> 
Nope, that's not a thing.  Changing the ebuild doesn't affect the checksum in the Manifest.  Either GitHub regenerated the tarball (happens rarely) or upstream reused the same tag.  Either way, please don't overload bugs with more than one unrelated issue next time.

**Comment by blshkv** [comment]
please try the latest ebuild, and open this bug report if issue till persist
https://github.com/pentoo/pentoo-overlay/pull/2751
